code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -12,50 +12,52 @@
 	if (!check_and_use_plasma_owner())
 		return
 
+	//X = xeno user, A = target atom
+	var/list/turf/target_turfs = getline2(X, A, include_from_atom = FALSE)
+	var/len = LAZYLEN(target_turfs)
+	if(len > 3)
+		target_turfs = target_turfs.Copy(1, 4)
+
 	// Get list of target mobs
 	var/list/target_mobs = list()
+	var/blocked = FALSE
 
-
-	var/list/target_turfs = list()
-	var/facing = Get_Compass_Dir(X, A)
-	var/turf/T = X.loc
-	var/turf/temp = X.loc
-
-	for(var/x in 0 to 2)
-		temp = get_step(T, facing)
-		if(!temp || temp.density || temp.opacity)
-			break
-
-		var/blocked = FALSE
-		for(var/obj/structure/S in temp)
-			if(istype(S, /obj/structure/window/framed))
-				var/obj/structure/window/framed/W = S
-				if(!W.unslashable)
-					W.deconstruct(disassembled = FALSE)
-
-			if(S.opacity)
-				blocked = TRUE
-				break
+	for(var/turf/path_turf as anything in target_turfs)
```

Comment:
I don't know if `as anything` is required here but I based this off of 
https://github.com/cmss13-devs/cmss13/blob/84d29ec1541f13eff680470cd948b5f9ffd9ee56/code/modules/mob/living/carbon/xenomorph/abilities/general_powers.dm#L933-L937

Reply:
for loops have an implicit type check, as anything means it won't type check. If the only thing in path is the turf then i'm fairly certain a type check is not required.